### PR TITLE
fix: docs generation runs outside of build matrix now

### LINKED ISSUE #161: Bug: release action - docs git push fails
### Describe the bug

Followup to #157 - looks like since the release process commits the changelog during the release flow in `semantic-release` we'll need to pull new changes.

```
Run just gen-docs
go run main.go docs ./docs/
Generating documentation in /home/runner/work/pizza-cli/pizza-cli/docs...
Finished generating documentation in /home/runner/work/pizza-cli/pizza-cli/docs
[beta d2cb7cf] chore: automated docs generation for release
 14 files changed, 17 insertions(+), 17 deletions(-)
To https://github.com/open-sauced/pizza-cli
 ! [rejected]        HEAD -> beta (fetch first)
error: failed to push some refs to 'https://github.com/open-sauced/pizza-cli'
hint: Updates were rejected because the remote contains work that you do not
hint: have locally. This is usually caused by another repository pushing to
hint: the same ref. If you want to integrate the remote changes, use
hint: 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

https://github.com/open-sauced/pizza-cli/actions/runs/10775149597/job/29878973449#step:5:16

---

Could we also make this co-authored by the original pusher like `semantic-release` has it?

Screenshot 2024-09-09 at 8 27 54 AM

### Steps to reproduce

N/a

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

This continues in the work in #162, however, now the docs is a separate job so it doesn't repeat itself in the build matrix job.

## Related Tickets & Documents

Fixes #161

## Steps to QA

We can try on beta again. This is a little more difficult to test on a test repo.

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

**Review by nickytonline — COMMENTED** [review]
I think this makes sense @open-sauced/engineering, It's a little more difficult to test all this on a test repo. I could probably fork and remove some of what semantic release and the build do to test this out.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -59,10 +59,46 @@ jobs:
...
+          go-version: 1.22.x
+
+      - name: "🤲 Setup Just"
+        uses: extractions/setup-just@v2
+
+      - name: "📗 Generate Documentation"
+        run: ./scripts/generate-docs.sh
+        env:
+          GITHUB_REF: ${{ github.ref }}
+          GH_TOKEN: ${{ github.token }}
```
Comment (brandonroberts) [inline .github/workflows/release.yaml:95]:
```suggestion
          GH_TOKEN: ${{ steps.generate_token.outputs.token }}
```

Comment (nickytonline) [reply .github/workflows/release.yaml]: Was reading up on it and basically `steps.generate_token.outputs.token` probably has more permissions to do things than `github.token`.

**Review by brandonroberts — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -59,10 +59,46 @@ jobs:
...
+      - release
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.OS_GITHUB_APP_ID }}
```
Comment (nickytonline) [inline .github/workflows/release.yaml:74]: I'm getting this warning in VS Code. I think it's OK, as I'm doing the same thing the release job is doing.

CleanShot 2024-09-09 at 15 17 52

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -59,10 +59,46 @@ jobs:
...
+          go-version: 1.22.x
+
+      - name: "🤲 Setup Just"
+        uses: extractions/setup-just@v2
+
+      - name: "📗 Generate Documentation"
+        run: ./scripts/generate-docs.sh
+        env:
+          GITHUB_REF: ${{ github.ref }}
+          GH_TOKEN: ${{ steps.generate_token.outputs.token }}
```
Comment (jpmcb) [inline .github/workflows/release.yaml:95]: `GH_TOKEN` doesn't actually get used in `scripts/generate-docs.sh`: shouldn't there be an extra step to authenticate with the token to git? I'm not sure what `semantic-release` is doing under the hood with it's `GITHUB_TOKEN` but I'd imagine we need to do something explicitly with it like:

```sh
git remote set-url origin "https://${GH_TOKEN}@github.com/open-sauced/pizza-cli"
```

Comment (nickytonline) [reply .github/workflows/release.yaml:95]: Gonna take a peek at this in the AM.

Comment (jpmcb) [reply .github/workflows/release.yaml:95]: > For the GH_TOKEN, I thought it was used under the hood

Ahhhhh that makes sense. Let's just send it: so hard to actually debug this stuff without seeing it in the actual action going through.

**Review by jpmcb — APPROVED** [review]

**Pull request merged**